### PR TITLE
`chat/restore`: set default llm for null `modelID` param

### PR DESCRIPTION
**Pull request by Sa1to**
It fixes https://github.com/sourcegraph/jetbrains/issues/620
## Test plan
Restore chat from history with null model 
(There is automated test in index.test.ts)

### CODE REVIEW

File: agent/src/index.test.ts [hunk 1]
```diff
@@ -302,6 +302,44 @@ describe('Agent', () => {
             )
         }, 30_000)
 
+        it('chat/restore (With null model)', async () => {
+            // Step 1: create a chat session where I share my name.
```
Comment (pkukielka) [inline agent/src/index.test.ts:306]: Incorrect comment?

File: agent/src/agent.ts [hunk 2]
```diff
@@ -710,7 +712,10 @@ export class Agent extends MessageHandler {
         })
 
         this.registerAuthenticatedRequest('chat/restore', async ({ modelID, messages, chatID }) => {
-            const chatModel = new SimpleChatModel(modelID, [], chatID)
+            const chatModel =
+                modelID === null
+                    ? new SimpleChatModel(ModelProvider.get(ModelUsage.Chat).at(0)!.model, [], chatID)
```
Comment (pkukielka) [inline agent/src/agent.ts:717]: Unfortunately looks like it's not so simple. Look at:
cody/vscode/src/models/index.ts, function `getModel`.
Looks like for pro users default model is the last one used. 
I think we can simplify a bit for now, but that logic (and logic in our client) won't be 100% correct :(

**Review by olafurpg — COMMENTED** [review]
@szymonprz FYI, you can run `./agent/scripts/reset-recordings-to-main.sh` to reset the recordings and start from scratch. The very large diff indicates that something may be wrong in the recordings. The recording file should only have additions for the newly added test case.

File: agent/src/agent.ts (hunk 2 above)
Comment (olafurpg) [inline agent/src/agent.ts:717]: It's always better to report a helpful error than to use `!`

```
const model = ModelProvider.get(ModelUsage.Chat).at(0)?.model
if (!model) { throw new Error( ... ) }
...
```

File: agent/src/agent.ts [hunk 3]
```diff
@@ -710,7 +712,10 @@ export class Agent extends MessageHandler {
         })
 
         this.registerAuthenticatedRequest('chat/restore', async ({ modelID, messages, chatID }) => {
-            const chatModel = new SimpleChatModel(modelID, [], chatID)
+            const chatModel =
+                modelID === null
```
Comment (olafurpg) [inline agent/src/agent.ts:716]: We should always handle both null and undefined
```suggestion
                modelID === null || modelID === undefined
```

It's normally easier to do do the condition like this
```
const foo = modelID ? ... : /* when null or undefined */
```

Comment (olafurpg) [reply agent/src/agent.ts:716]: Note that `modelID?: string` does not allow `null` values, you need to type it as `modelID?: string | null` to make it 100% correct. However, we don't do this consistently everywhere

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**